src/api/middleware/application.js

```diff
@@ -0,0 +1,19 @@
+import CompanyApplication, { CompanyApplicationRules } from "../../models/CompanyApplication.js";
+import { APIError, ErrorTypes } from "./errorHandler.js";
+import { StatusCodes as HTTPStatus } from "http-status-codes/build/cjs/status-codes.js";
+import { RECOVERY_LINK_EXPIRATION } from "../../models/constants/ApplicationStatus.js";
+
+export const exceededCreationTimeLimit = async (email) => {
+    const cursor = await CompanyApplication.findOne({ email, isVerified: false }).exec();
+    if (cursor !== null && Date.now() - cursor.submittedAt < RECOVERY_LINK_EXPIRATION * 1000) {
+        throw new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, CompanyApplicationRules.APPLICATION_RECENTLY_CREATED.msg);
+    }
+    return true;
+};
+
+export const deleteApplications = async (email) => {
+    await CompanyApplication.deleteMany({ email: email, isVerified: false }).catch(function(error) {
+        console.error(error);
+        throw (error); // Failure
```

Comment:
```suggestion
        throw error;
```

Reply:
This should be wrapped in a `try/catch` instead of using the built-in `catch` method of promises.